File: Sort an array of 0s, 1s and 2s - GFG/sort-an-array-of-0s-1s-and-2s.py
#User function Template for python3

class Solution:
    def sort012(self,nums,n):
        # code here
        zero, one, two = nums.count(0), nums.count(1), nums.count(2)
        index = 0
        for i in range(zero):
            nums[index] = 0
            index += 1
        for i in range(one):
            nums[index] = 1
            index += 1
        for i in range(two):
            nums[index] = 2
            index += 1



#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    t=int(input())
    for _ in range(t):
        n=int(input())
        arr=[int(x) for x in input().strip().split()]
        ob=Solution()
        ob.sort012(arr,n)
        for i in arr:
            print(i, end=' ')
        print()

# } Driver Code Ends

File: Sort an array of 0s, 1s and 2s - GFG/sort-an-array-of-0s-1s-and-2s.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;



// } Driver Code Ends
class Solution
{
    public:
    void sort012(int a[], int n){
        // code here 
        int zero = 0, one = 0, two = 0;
        for(int i=0; i<n; i++){
            if(a[i]==0)
                zero++;
            else if(a[i]==1)
                one++;
            else
                two++;
        }
        int i=0;
        while(zero > 0){
            a[i] = 0;
            zero--;
            i++;
        }
        while(one > 0){
            a[i] = 1;
            i++;
            one--;
        }
        while(two > 0){
            a[i] = 2;
            i++;
            two--;
        }
    }
    
};

//{ Driver Code Starts.
int main() {

    int t;
    cin >> t;

    while(t--){
        int n;
        cin >>n;
        int a[n];
        for(int i=0;i<n;i++){
            cin >> a[i];
        }

        Solution ob;
        ob.sort012(a, n);

        for(int i=0;i<n;i++){
            cout << a[i]  << " ";
        }

        cout << endl;
        
        
    }
    return 0;
}


// } Driver Code Ends